testapps/php56_e2e/web/grpc_pubsub.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+require_once __DIR__ . '/../vendor/autoload.php';
+
+use Google\Auth\ApplicationDefaultCredentials;
+
+function updateAuthMetadataCallback($context)
+{
+    $auth_credentials = ApplicationDefaultCredentials::getCredentials(
+        ['https://www.googleapis.com/auth/pubsub']
+    );
+    return $auth_credentials->updateMetadata($metadata = [],
+                                             $context->service_url);
+}
+
+$channel_credentials = Grpc\ChannelCredentials::createSsl(
+    file_get_contents(__DIR__ . '/../vendor/grpc/grpc/etc/roots.pem'));
+
+$opts = ['credentials' => $channel_credentials,
+         'grpc.ssl_target_name_override' => 'pubsub.googleapis.com'];
+
+$client = new google\pubsub\v1\PublisherClient('pubsub.googleapis.com', $opts);
+
+$deadline = Grpc\Timeval::InfFuture();
+
+$req = new google\pubsub\v1\ListTopicsRequest();
+$proj = getenv('GAE_LONG_APP_ID');
+$req->setProject('projects/' . $proj);
+$call = $client->ListTopics(
+    $req,
+    [],
+    ['call_credentials_callback' => 'updateAuthMetadataCallback']);
```

Comment:
hm, so this callback `'updateAuthMetadataCallback'` is meant to be seen in gRPC's perspective. It's as if the gRPC core library is saying, "I am ready for the auth credentials. Here's what we know about this call so far, contained in the `$context` variable (the most important one being `$context->service_url`, coz after load balancing, user may not know which is the host until now)". So with this `$context` variable, we ask the auth library for the auth header by calling `$auth_credentials->updateMetadata` passing `$context->service_url` to it. 

To me these 2 callbacks don't need to lineup because the outer gRPC callback can do more than just Auth. I think it;s OK to leave this as is.